test/com/linkedin/drelephant/schedulers/OozieSchedulerTest.java

```diff
@@ -0,0 +1,182 @@
+    assertEquals(0, scheduler.getWorkflowDepth());
+    assertEquals("oozie", scheduler.getSchedulerName());
+  }
+
+  @Test
+  public void testOozieLoadInfoWithMissingProperty() {
+    SchedulerConfigurationData schedulerConfig = makeConfig("oozie", getDefaultSchedulerParams());
+    OozieScheduler scheduler = new OozieScheduler("id", getNonOozieProperties(), schedulerConfig);
+
+    assertEquals(null, scheduler.getFlowDefUrl());
+    assertEquals(null, scheduler.getFlowDefId());
+    assertEquals(null, scheduler.getFlowExecUrl());
+    assertEquals(null, scheduler.getFlowExecId());
+
+    assertEquals(null, scheduler.getJobDefUrl());
+    assertEquals(null, scheduler.getJobDefId());
+    assertEquals(null, scheduler.getJobExecUrl());
+    assertEquals(null, scheduler.getJobExecId());
+
+    assertEquals(null, scheduler.getJobName());
+    assertEquals(0, scheduler.getWorkflowDepth());
+    assertEquals("oozie", scheduler.getSchedulerName());
+  }
+
+  @Test
+  public void testOozieLoadInfoWithNullProperty() {
+    SchedulerConfigurationData schedulerConfig = makeConfig("oozie", getDefaultSchedulerParams());
+    OozieScheduler scheduler = new OozieScheduler("id", null, schedulerConfig);
+
+    assertEquals(null, scheduler.getFlowDefUrl());
+    assertEquals(null, scheduler.getFlowDefId());
+    assertEquals(null, scheduler.getFlowExecId());
+    assertEquals(null, scheduler.getFlowExecUrl());
+
+    assertEquals(null, scheduler.getJobDefId());
+    assertEquals(null, scheduler.getJobDefUrl());
+    assertEquals(null, scheduler.getJobExecId());
+    assertEquals(null, scheduler.getJobExecUrl());
+
+    assertEquals(null, scheduler.getJobName());
+    assertEquals(0, scheduler.getWorkflowDepth());
+    assertEquals("oozie", scheduler.getSchedulerName());
+  }
+
+  @Test
+  public void testOozieLoadsNameFromConfData() {
+    SchedulerConfigurationData schedulerConfig = makeConfig("othername", getDefaultSchedulerParams());
+    OozieScheduler scheduler = new OozieScheduler("id", null, schedulerConfig);
+    assertEquals("othername", scheduler.getSchedulerName());
+  }
+
+  private static Properties getNonOozieProperties() {
+    return new Properties();
+  }
+
+  private static Properties getOozieProperties() {
+    Properties properties = new Properties();
+
+    properties.put("oozie.action.id", "0004167-160629080632562-oozie-oozi-W@some-action");
+    properties.put("oozie.job.id", "0004167-160629080632562-oozie-oozi-W");
+
+    return properties;
+  }
+
+  private static SchedulerConfigurationData makeConfig(String name, Map<String, String> params) {
+    return new SchedulerConfigurationData(name, null, params);
+  }
+
+  private static Map<String, String> getDefaultSchedulerParams() {
+    Map<String, String> paramMap = new HashMap<String, String>();
+
+    paramMap.put("oozie_api_url", "http://oozie.api/");
+    paramMap.put("oozie_job_url_template", "http://oozie/search?workflow=%s");
```

Comment:
I was always commenting only about the values to use in the tests (i.e. `http://oozie/search` -> `http://oozie-ui/search`), and not about changing the behaviour. sorry if I didn't make it clear enough.